cmd/libsnap-confine-private/string-utils.c

```diff
@@ -69,3 +70,29 @@ int sc_must_snprintf(char *str, size_t size, const char *format, ...)
 
 	return n;
 }
+
+void sc_string_append(char *buf, size_t buf_size, const char *str)
+{
+	// Set errno in case we die.
+	errno = 0;
+	if (buf == NULL) {
+		die("cannot append string: buffer is NULL");
+	}
+	if (str == NULL) {
+		die("cannot append string: string is NULL");
+	}
```

Comment:
I feel so silly not being able to write a correct strcat after so many tries. Thank you for guiding me.